### DISCUSSION TITLE
How to output the plan nicely

### DISCUSSION THREAD

**Discussion by gitPlatformTests** [gruntwork-io/knowledge-base#794] [How do I]
Hey,
I run the command terragrunt run-all plan --terragrunt-include-external-dependencies

This command output includes a lot of lines from various projects and so:
1. It is not easy to understand which plan is of what project
2. it includes more than just the plan

is there a way to get only the plans (assuming there were no errors) and to be able to identify easily

thanks

---

<ins datetime="2023-12-17T12:12:48Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110647">Tracked in ticket #110647</a></p>
</ins>

**Comment by gitPlatformTests** [comment]
I tried to do this. first create the plan file:
```
terragrunt run-all plan   \
  --terragrunt-include-dir 'path/to/project1' \
  --terragrunt-include-dir 'path/to/project2' \
  -out /tmp/plan
```

then show it:
```
terragrunt run-all show   \
  --terragrunt-include-dir 'path/to/project1' \
  --terragrunt-include-dir 'path/to/project2' \
  /tmp/plan
```

the show command sometimes fails for plugins error like
```
│ Error: Failed to load plugin schemas
│
│ Error while loading schemas for plugin components: Failed to obtain
│ provider schema: Could not load the schema for provider
│ registry.terraform.io/hashicorp/archive: failed to instantiate provider
│ "registry.terraform.io/hashicorp/archive" to obtain schema: unavailable
│ provider "registry.terraform.io/hashicorp/archive"..
```

when i try to do the same thing each time for one project alone and not all in one it succeeds

**Comment by gitsstewart** [comment]
@gitPlatformTests it looks like there are two questions in this issue? One appears to be a feature request and the other an error. Are they in fact two separate items or somehow related? If related, could you provide more context on how the two are related?

**Comment by lev-ok** [comment]
@gitPlatformTests 

> is there a way to get only the plans (assuming there were no errors) and to be able to identify easily

You can reduce the verbose log by setting the [debug level](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-log-level) appropriately, for example `--terragrunt-log-level error`. If you have a better idea of how to make logs more readable, please provide an example by creating a new feature request https://github.com/gruntwork-io/terragrunt/issues


> the show command sometimes fails for plugins error like
│ Error: Failed to load plugin schemas
│
│ Error while loading schemas for plugin components: Failed to obtain
│ provider schema: Could not load the schema for provider
│ registry.terraform.io/hashicorp/archive: failed to instantiate provider
│ "registry.terraform.io/hashicorp/archive" to obtain schema: unavailable
│ provider "registry.terraform.io/hashicorp/archive"..

Are you using a plugin cache directory? If yes, please disable it and try to run again, if the issue disappears, then most likely it is a bug. In this case, you also need to create a new issue, but before you do that, take a look at this issue https://github.com/gruntwork-io/terragrunt/issues/1212, it might be the same.

**Comment by denis256** [comment]
Hi,
I suspect it may be because of the absolute path `-out /tmp/plan` - each module writes same file, so try to simply specify the output file `-out plan` (it will generate the output file in its own directory)
